cloudformation/tests/test_cfn_template_rendering.py

```diff
@@ -96,3 +96,86 @@ def test_hit_substack_rendering(tmp_path):
 
     # Run format check
     assert cfn_formatter.check_formatting([str(rendered_file)], "yaml")
+
+
+def test_cw_dashboard_substack_rendering(tmp_path):
+    test_config = {
+        "ClusterConfigMetadata": {
+            "sections": {
+                "cluster": ["default"],
+                "dcv": ["dcv"],
+                "ebs": ["custom1", "custom2", "custom3", "custom4", "custom5"],
+                "efs": ["customfs"],
+                "fsx": ["fsx1"],
+                "raid": ["raidgp2"],
+                "scaling": ["custom"],
+                "vpc": ["default"],
+            }
+        },
+        "KeyName": "first_cluster",
+        "BaseOS": "alinux2",
+        "Scheduler": "slurm",
+        "MasterInstanceType": "t2.micro",
+        "MasterRootVolumeSize": "25",
+        "ComputeRootVolumeSize": "25",
+        "ProxyServer": "NONE",
+        "EC2IAMRoleName": "NONE",
+        "S3ReadResource": "NONE",
+        "S3ReadWriteResource": "NONE",
+        "EFA": "NONE",
+        "EphemeralDir": "/scratch",
+        "EncryptedEphemeral": "false",
+        "CustomAMI": "ami-07fb283f3083e0d00",
+        "PreInstallScript": "NONE",
+        "PreInstallArgs": "NONE",
+        "PostInstallScript": "NONE",
+        "PostInstallArgs": "NONE",
+        "ExtraJson": "{}",
+        "AdditionalCfnTemplate": "NONE",
+        "CustomChefCookbook": "NONE",
+        "IntelHPCPlatform": "false",
+        "ScaleDownIdleTime": "3",
+        "VPCId": "vpc-034545fa84d175bc0",
+        "MasterSubnetId": "subnet-021ceb759253eeb29",
+        "AccessFrom": "0.0.0.0/0",
+        "AdditionalSG": "NONE",
+        "ComputeSubnetId": "subnet-0e49d198bb500e672",
+        "ComputeSubnetCidr": "NONE",
+        "UsePublicIps": "true",
+        "VPCSecurityGroupId": "NONE",
+        "AvailabilityZone": "eu-west-1a",
+        "SharedDir": "vol1,vol2,vol3,vol4,vol5",
+        "EBSSnapshotId": "NONE,NONE,NONE,NONE,NONE",
+        "VolumeType": "gp2,io1,sc1,st1,io1",
+        "VolumeSize": "20,20,500,500,20",
+        "VolumeIOPS": "100,200,100,100,100",
+        "EBSEncryption": "false,false,false,false,false",
+        "EBSKMSKeyId": "NONE,NONE,NONE,NONE,NONE",
+        "EBSVolumeId": "NONE,NONE,NONE,NONE,NONE",
+        "NumberOfEBSVol": "5",
+        "EFSOptions": "efs,NONE,generalPurpose,NONE,NONE,false,bursting,NONE,Valid",
+        "RAIDOptions": "raid,1,2,gp2,20,100,false,NONE",
+        "FSXOptions": "/fsx,NONE,1200,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE,NONE",
+        "DCVOptions": "master,8443,0.0.0.0/0",
+        "CWLogOptions": "true,14",
+        "EC2IAMPolicies": "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy",  # FIXME could be removed, added auto
+        # ,arn:aws:iam::aws:policy/CloudWatchFullAccess", # FIXME has been removed, we should see what is required
+        "Architecture": "x86_64",
+        "Cores": "NONE,NONE",
+    }
+
+    env = Environment(loader=FileSystemLoader(".."))
```

Comment:
This code seems very similar to the previous one.

Could we use the same test with a parametrize fixture?
The parameters could be: yaml_file, config, cfn_lint_warnings

Reply:
Done, added a TODO in case we want to ignore W2001 errors (declared vars unused) as it could be useful if we test rendering when all sections are not created (Ex: no RAID volume but still RaidVolumeIDs as a parameter, need to ignore the W2001 warning)